### PR TITLE
check early for non-scalar default_fill_value

### PR DESCRIPTION
**Pull request by jbrockmendel**
Broken off of an in-progress division-by-zero branch

### CODE REVIEW

File: pandas/core/sparse/frame.py [hunk 1]
```diff
@@ -74,6 +75,8 @@ def __init__(
         dtype=None,
         copy=False,
     ):
+        if not is_scalar(default_fill_value):
```
Comment (WillAyd) [inline pandas/core/sparse/frame.py]: Is a dict acceptable here at all? Think we allow that for normal DataFrame fillna

File: pandas/tests/sparse/frame/test_frame.py [hunk 2]
```diff
@@ -136,6 +136,12 @@ def test_constructor(self, float_frame, float_frame_int_kind, float_frame_fill0)
 
         repr(float_frame)
 
+    def test_constructor_fill_value_scalar(self):
```
Comment (simonjayhawkins) [inline pandas/tests/sparse/frame/test_frame.py]:
```suggestion
    def test_constructor_fill_value_not_scalar_raises(self):
```

File: pandas/core/sparse/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/sparse/frame.py]: @TomAugspurger are there cases where non-scalar is accepted?  I was getting errors in another branch caused by zero-dim ndarray getting through here.

Comment (jreback) [reply pandas/core/sparse/frame.py]: I don't believe we allow this (nor should)

**Comment by jreback** [comment]
is there an issue for this? I guess currently this raises but its an incorrect description of the error?

**Comment by jbrockmendel** [comment]
> is there an issue for this? I guess currently this raises but its an incorrect description of the error?

ATM this raises once it gets passed to SparseArray.  The actual bug this fixes is the `fill_value = func(...)` on 471.

**Comment by jreback** [comment]
thanks

**Pull request merged**